### DISCUSSION TITLE
Questions regarding Cache data clearing using CAD

### DISCUSSION THREAD

**Discussion by HOLIKADEW** [Cookie-AutoDelete/Cookie-AutoDelete#1642] [Q&A / Support]
Okay, so let me tell you how I want to use the Cookie AutoDelete addon. I basically want each tab to work like a private tab. So, suppose I go to a random website; after I close the tab after 15 seconds, I want CAD to auto-delete everything related to the website, not only limited to just cookies but also local side data and other stuff from my browser, except when I whitelist it by clicking [this](https://imgur.com/a/Qx3BZfl)  

For now, I just downloaded the addon, enabled auto-clearing cookies in the settings, and disabled notifications. I’ve been using it like this for months.

I thought it should delete everything as soon as I leave the tab. Recently, I logged into the TrueCaller website, didn’t make any rules or whitelist it in the addon, closed the tab, and when I came back after 5 minutes, I wasn’t logged out. So, I realized that the addon only deletes cookies, not the site data or other stuff. Also, when I visited my browser's cookies and site data page, I saw many websites still had their local data stored, which I don’t want. [see screenshot](https://imgur.com/a/hb15AXs) also btw how is time.is able to store one cookie into my browser? I don't have any ruleset for the website [see](https://imgur.com/a/WCs0Oy4)  

So, my question is, how can I make the CAD addon delete all cookies along with all the site data when I close the tab, except for the ones I whitelist from [here](https://imgur.com/a/Qx3BZfl)? Don’t you think it should automatically do this? Because when I whitelist a site, it asks me (with a checkbox) if it should keep the cache and other stuff. So, shouldn’t it just delete the data from non-whitelisted sites otherwise, anyway?

In the addon settings, I found a section called "Other Browsing Data Cleanup Options." Under that, it says "Check to delete cache and other stuff." Should check [them](https://imgur.com/a/jHPNQYQ ) all? And if I do, will CAD delete the site data for all sites every time, regardless of the whitelist? And every time I visit a website, will it download all those caches and stuff again? I mean, Is the whitelist only for cookies, or does it clear everything like cache except for cookies? I want the whitelist to keep everything for a site, and for websites that aren’t whitelisted, I want it to clear everything.

It’s also very confusing why we have [that](https://imgur.com/a/jHPNQYQ) option when the whitelist already asks if we want to keep the site data or not. Shouldn’t it just delete the site data for non-whitelisted sites anyway? Why do we have that "Other Browsing Data Cleanup Options" section?

So basically I want the add-on to delete everything single thing, not just the cookies about a website unless I whitelist it myself.

**Comment by kennethtran93** [comment]
Seems you already created a support issue as well.  Since I have replied onto that one I will close this one.

Duplicate of #1643.
